File: java/src/main/java/codingdojo/Fizzbuzz3.java
package codingdojo;

import java.util.ArrayList;
import java.util.stream.IntStream;
import java.util.stream.Stream;

public class Fizzbuzz3 {

    public static void fizzbuzz() {
        calculateFizzbuzz().forEach(System.out::println);
    }

    static Stream<String> calculateFizzbuzz() {
        return IntStream.range(1, 101).mapToObj(
                Fizzbuzz3::nToFizzbuzz);
    }

    static String nToFizzbuzz(int n) {
        if (n % 3 == 0 && n % 5 == 0)
            return "FizzBuzz";
        else if (n % 3 == 0)
            return "Fizz";
        else if (n % 5 == 0)
            return "Buzz";
        else
            return n + "";
    }
}

File: kotlin/src/main/kotlin/codingdojo/Fizzbuzz3.kt
package codingdojo;

class Fizzbuzz3 {
    companion object {
        fun fizzbuzz() {
            calculateFizzbuzz().forEach(System.out::println);
        }

        fun calculateFizzbuzz(): List<String> {
            return (1..100).map { i -> Fizzbuzz3.nToFizzbuzz(i) };
        }

        fun nToFizzbuzz(n: Int): String {
            if (n % 3 == 0 && n % 5 == 0)
                return "FizzBuzz";
            else if (n % 3 == 0)
                return "Fizz";
            else if (n % 5 == 0)
                return "Buzz";
            else
                return "$n";
        }
    }
}
